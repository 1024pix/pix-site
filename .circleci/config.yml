---
version: 2.0

jobs:
  test:
    docker:
      - image: cimg/node:16.20.0
    steps:
      - checkout
      - run: cat package*.json > cachekey
      - restore_cache:
          keys:
            - pix-site-npm-{{ checksum "cachekey" }}
      - run: npm ci
      - save_cache:
          key: pix-site-npm-{{ checksum "cachekey" }}
          paths:
            - ~/.npm
      - run:
          name: Test
          command: |
            npm test
      - run:
          name: Lint
          command: |
            npm run lint

  e2e:
    docker:
      - image: mcr.microsoft.com/playwright:v1.30.0-focal
    resource_class: medium
    steps:
      - checkout
      - run: cat package*.json > cachekey
      - restore_cache:
          keys:
            - v2-pix-site-e2e-npm-{{ checksum "cachekey" }}
      - run: npm ci
      - save_cache:
          key: v2-pix-site-e2e-npm-{{ checksum "cachekey" }}
          paths:
            - ~/.npm
            - ~/.cache/ms-playwright
      - run:
          name: Run E2E tests
          environment:
            PLAYWRIGHT_JUNIT_OUTPUT_NAME: test-results/e2e-junit-results.xml
          command: npm run test:e2e:run -- --reporter=junit
      - store_test_results:
          path: test-results/

workflows:
  version: 2
  build_and_test:
    jobs:
      - test
      - e2e
