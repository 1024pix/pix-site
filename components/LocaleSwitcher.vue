<template>
  <div class="locale-switcher">
    <button
      ref="buttonRef"
      class="locale-switcher__button"
      @click="toggleLanguagesMenu()"
      @keyup.esc="isLanguagesMenuActive && toggleLanguagesMenu()"
    >
      <img :src="`/images/${localeProperties.icon}`" />
      <span>{{ localeProperties.name }}</span>
    </button>
    <ul
      v-if="isLanguagesMenuActive"
      ref="languagesMenuRef"
      class="locale-switcher__languages-menu"
    >
      <li>
        <span>International</span>
        <ul>
          <li>
            <a href="#" @click.prevent.stop="setLocale('fr')">Français</a>
          </li>
          <li>
            <a href="#" @click.prevent.stop="setLocale('en')">English</a>
          </li>
        </ul>
      </li>
      <li>
        <a href="#" @click.prevent.stop="setLocale('fr-be')">
          Fédération Wallonie-Bruxelles
        </a>
      </li>
      <li>
        <a href="#" @click.prevent.stop="setLocale('fr-fr')">France</a>
      </li>
    </ul>
  </div>
</template>

<script setup>
import { onClickOutside } from "@vueuse/core";

const { setLocale, localeProperties } = useI18n();

const buttonRef = ref(null);
const languagesMenuRef = ref(null);
const isLanguagesMenuActive = ref(false);

function toggleLanguagesMenu() {
  isLanguagesMenuActive.value = !isLanguagesMenuActive.value;
}

onClickOutside(
  languagesMenuRef,
  () => {
    toggleLanguagesMenu();
  },
  { ignore: [buttonRef] }
);
</script>

<style lang="scss">
.locale-switcher {
  position: relative;
}

.locale-switcher__button {
  display: flex;
  height: 1.5rem;
  margin-bottom: 2px;
  align-items: center;
  padding: 2px 3px;
  font-family: $font-roboto;
  font-size: 0.875rem;
  font-weight: $font-medium;
  color: $grey-60;
  letter-spacing: 0.008rem;
  line-height: 1.375rem;
  border: none;
  outline: inherit;
  background-size: 1.3rem;
  background-position: 5% 50%;
  cursor: pointer;

  &:hover {
    color: $blue;
  }

  img {
    margin-right: 8px;
  }
}

.locale-switcher__languages-menu {
  position: absolute;
  background: white;
  padding: 10px;
}
</style>
