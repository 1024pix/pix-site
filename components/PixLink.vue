<template>
  <component :is="linkComponent">
    <slot />
  </component>
</template>

<script>
import prismicDOM from 'prismic-dom'
import { SITES_PRISMIC_TAGS } from '~/services/available-sites'

export default {
  name: 'PixLink',
  props: {
    field: {
      required: true,
      type: Object,
      default: null,
    },
  },
  computed: {
    linkComponent() {
      if (!this.field) {
        return null
      }
      let template = ''
      let url = prismicDOM.Link.url(this.field, this.$prismic.linkResolver)

      url = removeHostIfCurrentSite(url, this.$i18n.locale)

      if (this.field.link_type === 'Document') {
        const localeURL = getLocaleURL(url, this.$i18n, this.localePath)
        template = `
          <nuxt-link to="${localeURL}" exact>
            <slot/>
          </nuxt-link>
        `
      } else if (isCurrentLocaleURL(url, this.$i18n)) {
        template = `
          <nuxt-link to="${url}" exact>
            <slot/>
          </nuxt-link>
        `
      } else {
        const target = this.field.target
          ? `target="'${this.field.target}'" rel="noopener"`
          : ''
        template = `
          <a href="${url}" ${target}>
            <slot/>
          </a>
        `
      }
      return { template }
    },
  },
}

export function isCurrentLocaleURL(url, $i18n) {
  if (!url.startsWith('/')) return false

  const locale = getPathLocale(url, $i18n) ?? $i18n.defaultLocale
  return locale === $i18n.locale
}

function getLocaleURL(url, $i18n, localePath) {
  if (!url.startsWith('/') || getPathLocale(url, $i18n)) return url
  return localePath(url)
}

export function getPathLocale(path, $i18n) {
  const localeCodesWithoutDefault = $i18n.localeCodes.filter(
    (code) => code !== $i18n.defaultLocale
  )

  const rootMatch = new RegExp(
    `^/(${localeCodesWithoutDefault.join('|')})$`
  ).exec(path)
  if (rootMatch != null) return rootMatch[1]

  const pageMatch = new RegExp(
    `^/(${localeCodesWithoutDefault.join('|')})/`
  ).exec(path)
  if (pageMatch != null) return pageMatch[1]

  return undefined
}

export function removeHostIfCurrentSite(url, locale) {
  try {
    const parsedURL = new URL(url)

    if (!isCurrentSiteURL(parsedURL, locale)) return url

    return parsedURL.pathname + parsedURL.search + parsedURL.hash
  } catch (e) {
    // relative URL
    return url
  }
}

function isCurrentSiteURL(url, locale) {
  return url.host === getCurrentSiteHost(locale)
}

function getCurrentSiteHost(locale) {
  if (process.env.SITE === SITES_PRISMIC_TAGS.PIX_PRO) return 'pro.pix.fr'
  if (locale === 'fr-fr') return 'pix.fr'
  return 'pix.org'
}
</script>
