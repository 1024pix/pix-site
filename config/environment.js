import { SITES_PRISMIC_TAGS } from '../services/available-sites'

export const config = {
  site: process.env.SITE,
  siteDomain: process.env.SITE_DOMAIN,
  isFrenchDomain: process.env.SITE_DOMAIN === 'pix.fr',
  isPixSite: process.env.SITE === SITES_PRISMIC_TAGS.PIX_SITE,
  prismic: {
    apiEndpoint: process.env.PRISMIC_API_ENDPOINT,
  },
  matomo: {
    containerUrl: process.env.MATOMO_CONTAINER,
    debug: process.env.MATOMO_DEBUG || false,
  },
  featureToggles: {
    disableLanguageSwitcherPixProFr:
      process.env.FT_DISABLE_PIX_PRO_LANGUAGE_SWITCHER === 'true' &&
      process.env.SITE === SITES_PRISMIC_TAGS.PIX_PRO &&
      process.env.SITE_DOMAIN === 'pix.fr',
  },
}
