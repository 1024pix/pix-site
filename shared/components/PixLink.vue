<template>
  <component :is="linkComponent">
    <slot />
  </component>
</template>

<script>
export default {
  name: 'PixLink',
  props: {
    href: {
      required: true,
      type: String,
      default: '',
    },
  },
  computed: {
    linkComponent() {
      let template = ''
      if (this.isOnCurrentDomain(this.href)) {
        template = `
          <nuxt-link to="${this.href}">
            <slot />
          </nuxt-link>
         `
      } else {
        template = `
          <a href="${this.href}">
            <slot />
          </a>
        `
      }
      return { template }
    },
  },
  methods: {
    isOnCurrentDomain(targetDomain) {
      return targetDomain.startsWith('/') && !targetDomain.startsWith('//')
    },
  },
}
</script>
