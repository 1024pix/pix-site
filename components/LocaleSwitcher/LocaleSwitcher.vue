<template>
  <div
    v-if="type === 'with-dropdown'"
    class="locale-switcher"
    @keydown.esc="hideMenu"
  >
    <button
      class="locale-switcher__button"
      :aria-label="$t('locale-switcher-label')"
      aria-haspopup="true"
      :aria-expanded="showMenu.toString()"
      @click.stop.prevent="toggleMenu()"
    >
      {{ $t(currentLocaleCode) }}
      <fa
        v-if="showMenu"
        icon="angle-up"
        class="locale-switcher-button__angle"
      />
      <fa v-else icon="angle-down" class="locale-switcher-button__angle" />
    </button>
    <ul v-if="showMenu" class="locale-switcher__dropdown-menu">
      <li
        v-for="(option, index) in languages.menu"
        :key="`locale-switcher-${index}`"
        :class="{
          'locale-switcher__dropdown-menu--active':
            option.localeCode === currentLocaleCode,
        }"
      >
        <div class="locale-switcher__locale">
          <template v-if="option.children">
            <button
              v-if="option.children"
              class="locale-switcher-sub-menu-button"
              @click.stop.prevent="toggleSubMenu()"
            >
              <img
                class="locale-switcher__img"
                :src="'/images/' + option.icon"
                alt=""
              />
              <div class="locale__text">{{ $t(option.name) }}</div>
              <fa
                icon="angle-right"
                class="locale-switcher__icon locale-switcher-button__angle"
              />
            </button>
            <locale-switcher-sub-menu
              :show-sub-menu="showSubMenu"
              :available-languages="option.children"
              :current-locale-code="currentLocaleCode"
              :selected-menu="selectedMenu"
            />
          </template>
          <pix-link v-else-if="!option.children" :href="getIndexUrl(option)">
            <img
              class="locale-switcher__img"
              :src="'/images/' + option.icon"
              alt=""
            />
            <div class="locale__text">
              <div>{{ $t(option.name) }}</div>
              <div v-if="option.subtitle" class="locale-switcher__subtitle">
                {{ $t(option.subtitle) }}
              </div>
            </div>
          </pix-link>
        </div>
      </li>
    </ul>
    <span class="separator" />
  </div>
  <div v-else-if="type === 'only-text'">
    <ul class="locale-switcher-burger-menu">
      <li v-for="option in languages.menu" :key="option.key">
        <template v-if="option.children">
          <span
            v-for="child in option.children"
            :key="child.name"
            :class="{
              'locale-switcher-burger-menu__locale': true,
              'locale-switcher-burger-menu--active':
                child.localeCode === currentLocaleCode,
            }"
          >
            <pix-link :href="getIndexUrl(child)">
              {{ $t(option.name) }} - {{ $t(child.name) }}
            </pix-link>
            <br />
          </span>
        </template>
        <template v-if="!option.children">
          <span
            :class="{
              'locale-switcher-burger-menu__locale': true,
              'locale-switcher-burger-menu--active':
                option.localeCode === currentLocaleCode,
            }"
          >
            <pix-link :href="getIndexUrl(option)">
              {{ $t(option.name) }}
            </pix-link>
          </span>
        </template>
      </li>
    </ul>
  </div>
</template>

<script>
import { language } from '~/config/language'
import { getAbsoluteUrlIfSwitchWebsite } from '~/services/get-absolute-url-if-switch-website'

export default {
  name: 'LocaleSwitcher',
  props: {
    type: {
      type: String,
      default: 'with-dropdown',
    },
  },
  data() {
    const languages = language
    return {
      showMenu: false,
      showSubMenu: false,
      languages,
    }
  },
  computed: {
    currentLocaleCode() {
      return this.$i18n.locale || this.$i18n.defaultLocale
    },
    selectedMenu() {
      return this.languages.menu
        .flatMap((menuItem) => menuItem.children ?? menuItem)
        .find((locale) => locale.localeCode === this.currentLocaleCode)
    },
  },
  mounted() {
    const page = document.getElementsByTagName('body')[0]

    page.addEventListener('click', this.hideMenu)
  },
  beforeDestroy() {
    const page = document.getElementsByTagName('body')[0]
    page.removeEventListener('click', this.hideMenu)
  },
  methods: {
    toggleMenu() {
      this.showMenu = !this.showMenu
    },
    toggleSubMenu() {
      this.showSubMenu = !this.showSubMenu
    },
    hideMenu() {
      this.showMenu = false
      this.showSubMenu = false
    },
    getIndexUrl(menuItem) {
      const locale = language.locales.find(
        (locale) => locale.code === menuItem.localeCode
      )
      return getAbsoluteUrlIfSwitchWebsite({
        relativeTarget: locale.code === 'fr-fr' ? '/' : `/${locale.code}`,
        targetDomain: locale.domain,
      })
    },
  },
}
</script>

<style lang="scss">
.locale-switcher {
  display: none;
}
@include device-is('large-screen') {
  .locale-switcher {
    display: flex;

    .separator {
      margin-left: 15px;
      border-left: 1px solid $grey-20;
      display: block;
      height: 1.5rem;
    }
    &__button {
      font-family: $font-roboto;
      font-size: 0.875rem;
      font-weight: $font-medium;
      color: $grey-60;
      letter-spacing: 0.008rem;
      line-height: 1.375rem;
      margin-bottom: 2px;
      border: none;
      outline: inherit;
      height: 1.5rem;
      background-size: 1.3rem;
      background-position: 5% 50%;
      padding: 2px 3px 0 30px;
      cursor: pointer;
      background: url('/images/globe.svg') no-repeat left;
      &:hover {
        color: $blue;
      }
    }
    &__dropdown-menu {
      position: absolute;
      z-index: 5;
      float: left;
      min-width: 150px;
      padding: 0;
      margin: 35px 0 0 0;

      list-style: none;
      font-size: 14px;
      text-align: left;
      background-color: $white;
      border: 1px solid $grey-20;
      border-radius: 4px;
      box-shadow: 0 6px 12px rgba(0, 0, 0, 0.175);
      background-clip: padding-box;

      .child {
        top: -1px;
        left: 100%;
        margin: 0;
        padding: 0;
        box-shadow: 0 0;
        display: block;
        min-height: 25px;
        .locale-switcher__locale > a {
          line-height: 23px;
        }
      }

      & > li {
        list-style: none;
        width: 100%;
        position: relative;
        margin: 0;
        padding: 0;
        &::before {
          content: '';
          margin: 0;
        }
        &:hover {
          background: $grey-10;
          .child {
            display: block;
          }
        }
      }
      & > li > a,
      & > li > span {
        display: block;
        line-height: 1.6rem;
        color: $grey-20;
        white-space: nowrap;
        text-decoration: none;
      }
      & > li > a:hover,
      & > li > span {
        background: $grey-10;
      }
      & > li:not(:last-child) {
        border-bottom: 1px solid $grey-20;
      }
      &--active {
        color: $grey-30;
        pointer-events: none;
        a {
          color: white;
        }
      }
    }
    &__locale {
      min-height: 25px;
      a,
      span,
      button {
        padding: 10px;
        font-family: $font-roboto;
        font-size: 0.9rem;
        font-weight: $font-medium;
        letter-spacing: 0.008rem;
        line-height: 1;
        color: inherit;
        display: flex;
        justify-content: space-between;
        align-items: center;
        &:hover {
          color: $blue;
          background: $grey-10;
        }
        .locale {
          &__text {
            flex: 1;
          }
        }
      }
      button {
        width: 100%;
        border: none;
        background: white;
        &:hover {
          color: $blue;
          background: $grey-10;
        }
        .locale {
          &__text {
            flex: 1;
            text-align: left;
          }
        }
      }
    }
    &__icon {
      font-size: large;
      float: right;
    }
    &__subtitle {
      font-size: smaller;
    }
    &__img {
      margin-right: 10px;
      height: 23px;
      width: 23px;
    }
  }
  .locale-switcher-button {
    &__angle {
      margin-left: 8px;
    }
  }
}

.locale-switcher-burger-menu {
  margin: 0 0 50px 0;
  font-size: 1rem;
  line-height: 1.5rem;
  color: $grey-30;
  list-style: none;
  ul {
    padding: 0;
  }
  li {
    list-style-type: none;
    padding-left: 0;
  }
  li::before {
    content: '';
    margin: 0;
  }
  &__locale {
    a {
      color: $grey-80;
      &:visited,
      &:focus,
      &:active,
      &:hover {
        color: $grey-80;
        text-decoration: underline;
      }
    }
  }
  &--active {
    pointer-events: none;
    a {
      color: $grey-25 !important;
    }
  }
}
</style>
