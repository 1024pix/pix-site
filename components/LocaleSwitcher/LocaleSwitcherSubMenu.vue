<template>
  <ul v-if="showSubMenu" class="locale-switcher__dropdown-menu child">
    <li
      v-for="availableLanguage in availableLanguages"
      :key="availableLanguage.key"
      :class="{
        'locale-switcher__dropdown-menu--active':
          availableLanguage.localeCode === currentLocaleCode,
      }"
    >
      <div class="locale-switcher__locale">
        <pix-link :href="getIndexUrl(availableLanguage)">
          {{ $t(availableLanguage.name) }}
        </pix-link>
      </div>
    </li>
  </ul>
</template>

<script>
import { getAbsoluteUrlIfSwitchWebsite } from '~/services/get-absolute-url-if-switch-website'
import { language } from '~/config/language'

export default {
  name: 'LocaleSwitcherSubMenu',
  props: {
    showSubMenu: {
      type: Boolean,
      default: false,
    },
    availableLanguages: {
      type: Array,
      default: null,
    },
    currentLocaleCode: {
      type: String,
      default: 'fr-fr',
    },
    selectedMenu: {
      type: Object,
      default: () => {},
    },
  },
  methods: {
    getIndexUrl(menuItem) {
      const locale = language.locales.find(
        (locale) => locale.code === menuItem.localeCode
      )
      return getAbsoluteUrlIfSwitchWebsite({
        relativeTarget: `/${locale.code}`,
        targetDomain: locale.domain,
      })
    },
  },
}
</script>
