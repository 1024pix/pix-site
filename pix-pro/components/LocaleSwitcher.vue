<template>
  <client-only>
    <div class="locale-switcher">
      <button
        ref="buttonRef"
        class="locale-switcher__button"
        :aria-label="t('locale-switcher.button.main-label')"
        aria-haspopup="menu"
        :aria-expanded="isLocalesMenuVisible"
        @click="toggleLocalesMenu"
        @keyup.esc="isLocalesMenuVisible && toggleLocalesMenu()"
      >
        <img :src="`/images/${localeProperties.icon}`" alt="" />
        <span>{{ localeProperties.name }}</span>
      </button>
      <ul v-show="isLocalesMenuVisible" ref="localesMenuRef" class="locale-switcher__languages-menu">
        <li>
          <button
            class="sub-menu-toggler"
            :aria-label="t('locale-switcher.button.international-label')"
            @click="toggleInternationalLanguages"
          >
            <img :src="`/images/${frLocale.icon}`" alt="" />
            <span>{{ t('locale-switcher.locales.international') }}</span>
          </button>
          <ul v-show="isInternationalLocalesVisible" class="sub-menu">
            <li v-if="frLocale" :class="{ active: localeProperties.code === frLocale.code }">
              <a
                :href="`${domainOrg || ''}/fr`"
                :aria-current="localeProperties.code === frLocale.code && 'page'"
                @click="updateLocaleCookie(frLocale.code)"
              >
                {{ frLocale.name }}
              </a>
            </li>
            <li v-if="enLocale" :class="{ active: localeProperties.code === enLocale.code }">
              <a
                :href="`${domainOrg || ''}/en`"
                :aria-current="localeProperties.code === enLocale.code && 'page'"
                @click="updateLocaleCookie(enLocale.code)"
              >
                {{ enLocale.name }}
              </a>
            </li>
          </ul>
        </li>
        <li v-if="frFrLocale" :class="{ active: localeProperties.code === frFrLocale.code }">
          <a :href="`${domainFr || ''}/`" :aria-current="localeProperties.code === frFrLocale.code && 'page'">
            <img :src="`/images/${frFrLocale.icon}`" alt="" />
            <span>{{ frFrLocale.name }}</span>
          </a>
        </li>
      </ul>
    </div>
  </client-only>
</template>

<script setup>
import { reachableLocales } from '../i18n.config';
import { onClickOutside } from '@vueuse/core';

const { getEnvironmentUrl } = useEnvironmentUrl();
const { setLocaleCookie } = useLocaleCookie();

const { localeProperties, locales, t } = useI18n();

const frLocale = reachableLocales.find((l) => l.code === 'fr');
const enLocale = reachableLocales.find((l) => l.code === 'en');
const frFrLocale = reachableLocales.find((l) => l.code === 'fr-fr');

const buttonRef = ref(null);
const localesMenuRef = ref(null);
const isLocalesMenuVisible = ref(false);
const isInternationalLocalesVisible = ref(false);

const config = useAppConfig();
const domainFr = config.domainFr;
const domainOrg = config.domainOrg;

function toggleLocalesMenu() {
  isLocalesMenuVisible.value = !isLocalesMenuVisible.value;

  if (!isLocalesMenuVisible.value) {
    isInternationalLocalesVisible.value = false;
  }
}

function toggleInternationalLanguages() {
  isInternationalLocalesVisible.value = !isInternationalLocalesVisible.value;
}

function updateLocaleCookie(localeCode) {
  setLocaleCookie(localeCode);
}

onClickOutside(
  localesMenuRef,
  () => {
    if (isLocalesMenuVisible.value) {
      toggleLocalesMenu();
    }
  },
  { ignore: [buttonRef] },
);
</script>

<style lang="scss">
.locale-switcher {
  display: none;
}

@include device-is('large-screen') {
  .locale-switcher {
    display: block;
    position: relative;
  }
}

.locale-switcher__button {
  display: flex;
  align-items: center;
  height: 1.5rem;
  padding: 2px 3px;
  font-family: $font-roboto;
  font-size: 0.875rem;
  font-weight: $font-medium;
  color: $grey-60;
  letter-spacing: 0.008rem;
  line-height: 1.375rem;
  border: none;
  background-color: transparent;
  background-size: 1.3rem;
  background-position: 5% 50%;
  cursor: pointer;

  &:hover {
    color: $blue;
  }

  &::after {
    content: '';
    width: 0.5em;
    height: 0.5em;
    margin-left: 0.75em;
    border-bottom: 2px solid currentColor;
    border-right: 2px solid currentColor;
    transform: translateY(-0.15em) rotate(45deg);
  }

  img {
    margin-right: 8px;
  }
}

.locale-switcher__languages-menu {
  top: calc(100% + 0.5rem);

  &,
  .sub-menu {
    position: absolute;
    z-index: 1;
    list-style: none;
    margin: 0;
    padding: 0;
    background: white;
    border: 1px solid $grey-20;
    border-radius: 0.25rem;
  }

  li {
    padding: 0;
    font-size: 0.9rem;
    color: $grey-60;

    &:hover {
      background-color: $grey-10;
      color: $blue;
    }

    &.active {
      color: $grey-30;
      pointer-events: none;
    }

    &::before {
      content: none;
    }

    & + li {
      border-top: 1px solid $grey-20;
    }

    & > *:not(.sub-menu) {
      display: flex;
      align-items: center;
      min-width: max-content;
      padding: 0.75em;
      line-height: 1;
      color: inherit;

      img {
        width: 1.5em;
        margin-right: 0.75em;
      }

      span {
        flex-grow: 1;
        white-space: nowrap;
      }
    }
  }

  .sub-menu {
    top: -1px;
    left: 100%;
    min-width: 150px;
  }

  .sub-menu-toggler {
    align-items: center;
    width: 100%;
    appearance: none;
    background: transparent;
    border: none;
    text-align: left;
    font: inherit;
    cursor: pointer;

    &::after {
      content: '';
      display: block;
      width: 0.6em;
      height: 0.6em;
      margin-left: auto;
      border-bottom: 2px solid currentColor;
      border-right: 2px solid currentColor;
      transform: translateY(-5%) rotate(-45deg);
    }

    span {
      margin-right: 0.65em;
    }
  }
}
</style>
