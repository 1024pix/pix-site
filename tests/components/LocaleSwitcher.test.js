import { shallowMount, mount } from '@vue/test-utils'
import { config } from '~/config/environment'
import PixLink from '~/components/PixLink'
import LocaleSwitcher from '~/components/LocaleSwitcher/LocaleSwitcher.vue'
import LocaleSwitcherSubMenu from '~/components/LocaleSwitcher/LocaleSwitcherSubMenu.vue'

jest.mock('~/config/environment', () => ({
  config: {
    site: 'pix-site',
    isPixSite: true,
    domain: {
      french: 'example.fr',
      international: 'example.org',
    },
  },
}))

LocaleSwitcherSubMenu.components = { 'pix-link': PixLink }

describe('Component: LocaleSwitcher', () => {
  describe('when site is pix-pro', () => {
    beforeEach(() => {
      config.site = 'pix-pro'
      config.isPixSite = false
    })

    it('should render the locale switcher', () => {
      // given
      const $t = () => {}
      const $i18n = { locale: 'fr-fr' }

      // when
      const wrapper = shallowMount(LocaleSwitcher, {
        stubs: {
          LocaleSwitcherSubMenu: true,
          fa: true,
        },
        propsData: {
          type: 'with-dropdown',
        },
        data: LocaleSwitcher.data,
        computed: LocaleSwitcher.computed,
        mocks: { $t, $i18n },
      })

      // then
      expect(wrapper.find('.locale-switcher__button').exists()).toBe(true)
    })

    it('should render the locale switcher correctly for pix-pro', async () => {
      // given
      const $t = () => {}
      const $i18n = { locale: 'fr-fr' }

      // when
      const wrapper = mount(LocaleSwitcher, {
        components: { LocaleSwitcherSubMenu, PixLink },
        mocks: { $t, $i18n },
        stubs: ['fa'],
      })
      await wrapper.find('button').trigger('click')
      await wrapper.find('.locale-switcher-sub-menu-button').trigger('click')

      // then
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('when site is pix-site', () => {
    beforeEach(() => {
      config.site = 'pix-site'
      config.isPixSite = true
    })

    it('should render the locale switcher correctly for pix-site', async () => {
      // given
      const $t = () => {}
      const $i18n = { locale: 'fr-fr' }

      // when
      const wrapper = mount(LocaleSwitcher, {
        components: { LocaleSwitcherSubMenu, PixLink },
        mocks: { $t, $i18n },
        stubs: ['fa'],
      })
      await wrapper.find('button').trigger('click')
      await wrapper.find('.locale-switcher-sub-menu-button').trigger('click')

      // then
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe.only('#isInternational', () => {
    describe('when locale is international', () => {
      it('returns true', () => {
        // given
        const $t = () => {}
        const $i18n = { locale: 'fr-fr' }
        const locale = {
          localeCode: 'fr',
        }
        const wrapper = mount(LocaleSwitcher, {
          components: { PixLink },
          mocks: { $t, $i18n },
          stubs: ['fa'],
        })

        // when
        const result = wrapper.vm.isInternational(locale)

        // then
        expect(result).toBe(true)
      })
    })
    describe('when locale is not international', () => {
      it('returns false', () => {
        // given
        const $t = () => {}
        const $i18n = { locale: 'fr-fr' }
        const locale = {
          localeCode: 'fr-fr',
        }
        const wrapper = mount(LocaleSwitcher, {
          components: { PixLink },
          mocks: { $t, $i18n },
          stubs: ['fa'],
        })

        // when
        const result = wrapper.vm.isInternational(locale)

        // then
        expect(result).toBe(false)
      })
    })
  })
})
