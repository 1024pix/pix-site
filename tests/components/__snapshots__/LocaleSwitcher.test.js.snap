// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LocaleSwitcher when site is pix-pro should render the locale switcher correctly for pix-pro 1`] = `
<div
  class="locale-switcher"
>
  <button
    aria-expanded="true"
    aria-haspopup="true"
    class="locale-switcher__button"
    style="background: url(/images/flag-fr.svg) no-repeat left;"
  >
    
    
    
    <fa-stub
      class="locale-switcher-button__angle"
      icon="angle-up"
    />
  </button>
   
  <ul
    class="locale-switcher__dropdown-menu"
  >
    <li
      class=""
    >
      <div
        class="locale-switcher__locale"
      >
        <button
          class="locale-switcher-sub-menu-button"
        >
          <img
            alt=""
            class="locale-switcher__img"
            src="/images/globe-europe.svg"
          />
           
          <div
            class="locale__text"
          >
            
          </div>
           
          <fa-stub
            class="locale-switcher__icon locale-switcher-button__angle"
            icon="angle-right"
          />
        </button>
         
        <ul
          class="locale-switcher__dropdown-menu child"
        >
          <li
            class=""
          >
            <div
              class="locale-switcher__locale"
            >
              <a
                href="http://example.org/fr"
              >
                
        
      
              </a>
            </div>
          </li>
          <li
            class=""
          >
            <div
              class="locale-switcher__locale"
            >
              <a
                href="http://example.org/en"
              >
                
        
      
              </a>
            </div>
          </li>
        </ul>
      </div>
    </li>
    <li
      class=""
    >
      <div
        class="locale-switcher__locale"
      >
        <a
          href="http://example.org/fr-be"
        >
          <img
            alt=""
            class="locale-switcher__img"
            src="/images/flag-be.svg"
          />
           
          <div
            class="locale__text"
          >
            <div>
              
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      class="locale-switcher__dropdown-menu--active"
    >
      <div
        class="locale-switcher__locale"
      >
        <a
          href="http://example.fr/"
        >
          <img
            alt=""
            class="locale-switcher__img"
            src="/images/flag-fr.svg"
          />
           
          <div
            class="locale__text"
          >
            <div>
              
            </div>
          </div>
        </a>
      </div>
    </li>
  </ul>
   
  <span
    class="separator"
  />
</div>
`;

exports[`Component: LocaleSwitcher when site is pix-site should render the locale switcher correctly for pix-site 1`] = `
<div
  class="locale-switcher"
>
  <button
    aria-expanded="true"
    aria-haspopup="true"
    class="locale-switcher__button"
    style="background: url(/images/flag-fr.svg) no-repeat left;"
  >
    
    
    
    <fa-stub
      class="locale-switcher-button__angle"
      icon="angle-up"
    />
  </button>
   
  <ul
    class="locale-switcher__dropdown-menu"
  >
    <li
      class=""
    >
      <div
        class="locale-switcher__locale"
      >
        <button
          class="locale-switcher-sub-menu-button"
        >
          <img
            alt=""
            class="locale-switcher__img"
            src="/images/globe-europe.svg"
          />
           
          <div
            class="locale__text"
          >
            
          </div>
           
          <fa-stub
            class="locale-switcher__icon locale-switcher-button__angle"
            icon="angle-right"
          />
        </button>
         
        <ul
          class="locale-switcher__dropdown-menu child"
        >
          <li
            class=""
          >
            <div
              class="locale-switcher__locale"
            >
              <a
                href="http://example.org/fr"
              >
                
        
      
              </a>
            </div>
          </li>
          <li
            class=""
          >
            <div
              class="locale-switcher__locale"
            >
              <a
                href="http://example.org/en"
              >
                
        
      
              </a>
            </div>
          </li>
        </ul>
      </div>
    </li>
    <li
      class=""
    >
      <div
        class="locale-switcher__locale"
      >
        <a
          href="http://example.org/fr-be"
        >
          <img
            alt=""
            class="locale-switcher__img"
            src="/images/flag-be.svg"
          />
           
          <div
            class="locale__text"
          >
            <div>
              
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      class="locale-switcher__dropdown-menu--active"
    >
      <div
        class="locale-switcher__locale"
      >
        <a
          href="http://example.fr/"
        >
          <img
            alt=""
            class="locale-switcher__img"
            src="/images/flag-fr.svg"
          />
           
          <div
            class="locale__text"
          >
            <div>
              
            </div>
          </div>
        </a>
      </div>
    </li>
  </ul>
   
  <span
    class="separator"
  />
</div>
`;
